### PR TITLE
feat: optional tether prop for DropdownTrigger

### PR DESCRIPTION
**Pull request by hannahstwinter**
Introduces prop tether with value of 'right' or 'left' (default 'left') to DropdownTrigger.
Value of 'right' will tether the dropdown menu to the right of the trigger.

### CODE REVIEW

File: src/Dropdown.jsx [hunk 1]
```diff
@@ -9,18 +9,22 @@ const DROPDOWN_TYPES = {
...
 class Dropdown extends React.Component {
   render() {
-    let style, classes;
+    let dropdownStyle, menuStyle, classes;
 
-    style = { float: 'left' };
+    dropdownStyle = { float: 'left' };
+
+    if (this.props.tether) {
+      menuStyle = { position: 'static' };
```
Comment (hannahstwinter) [inline src/Dropdown.jsx]: Absolute positioned elements with absolute positioned children do not change size in relation to their children. The _containerDiv created in the DropdownTrigger was not expanding to the size of the menu. In accord, the container div had a size of 0 and was positioned at the top left corner of the menu, so tethering on any corner of the container div would always translate to the top left corner of the menu.

**Comment by willcumbie** [comment]
I'm not crazy about the property name "tether." What about "position" or "alignment"?

**Comment by adampickeral** [comment]
i would rename the `tether` prop to `placement`.

**Comment by hannahstwinter** [comment]
I could go for alignment - it would read better against the concept of Tether in the component, the Tether instance, etc.

**Comment by adampickeral** [comment]
`placement` is what the popover uses

**Comment by hannahstwinter** [comment]
I worry 'placement' of right might convey that the menu should be placed to the right of the button (implying the tether point is on the left).

**Comment by adampickeral** [comment]
`alignment` works for me, makes sense in this case. the dropdown is aligned to the right or left of the trigger

File: src/Dropdown.jsx (hunk 1 above)
Comment (jdrush89) [inline src/Dropdown.jsx]: does it hurt to always have this style?

Comment (hannahstwinter) [inline src/Dropdown.jsx]: @jdrush89 I don't know... Also, I'm not 100% certain how to test that thoroughly enough to feel ok with making the change universal. 😿

**Comment by adampickeral** [comment]
LGTM!

File: src/Dropdown.jsx (hunk 1 above)
Comment (jdrush89) [inline src/Dropdown.jsx]: I think we're only using this in a few places in reach, I can help test when you bump the version if you want.  Without it we're relying on the 0 width container defaulting to the top left and I don't know if that behavior could change.

**Comment by wpwood** [comment]
LGTM2

🌸 🍶

**Pull request merged**